tests/roundtrip.rs

```diff
@@ -0,0 +1,144 @@
+             .read();//.unwrap();
+        match run_status {
+            Ok(o)=> {println!("{}", o)}
+            Err(e)=> {
+                panic!("Failed to archive files in {:?}: Error: {}",
+                dir.path().to_owned().to_str(), e)
+            }
+        }
+
+
+        //NOTE: must use .start() or test will hang indefinitely
+        // since Result is only generated after server process
+        // terminates for .run()
+        let process_handle;
+        let server_status = cmd!("./target/debug/static-filez",
+             "serve", "-p", "3000", &archive_path)
+             .start();
+             match server_status{
+                 Ok(o) => {println!("started server successfully."); process_handle = o;}
+                 Err(e) => {panic!("unable to start archive hosting server: {}", e)}
+             }
+
+           // assert!(false, "killed test for debug");
+
+        for i in 1 .. 100 {
+           // let mut tmpfile = tempfile::tempfile_in(archive_dir.path())
+                //.expect("failed to create temp file for zip file download");
+            let mut tmpfile_path = format!("{}/zip{}", archive_dir_path, i);
+            let mut tmpfile= File::create(tmpfile_path).expect("failed to create file to hold archived info");
+            let mut unzipped_file_path= format!("{}/unZipped_file{}", &archive_dir_path, i).to_owned();
+            //let x = unzipped_file_path.clone();
+            //assert!(false, x);
+
+            let url = String::from(format!("http://127.0.0.1:3000/file{}",i));
+
+
+            let mut unzipped_tempfile=File::create(&unzipped_file_path).unwrap();
+
+            let mut response = reqwest::get(&url).expect(&format!("failed to get file{} from server",i));
+            io::copy(&mut response, &mut tmpfile).expect(
+                &format!("failed to copy data for file{} from server response",i)
+                );
+
+            //while true{};
+            //TODO: failing to extract server zipped file to another file:
+            let extracter = unzip::Unzipper::new(tmpfile, &unzipped_file_path);
+            let unzip_result = extracter.unzip().expect("failed to unzip file");
+            let mut unzipped_contents = String::new();
+            let res= std::fs::read_to_string(&unzipped_file_path);
+            match res {
+                Ok(o) => {unzipped_contents=o.clone()}
+                Err(e) => {panic!("could not check unzipped file #{}: {}", i, e)}
+            }
+            //debug: leave test running so you can go check temp dir contents
+            //while true{};
+            //assert!(false, "unzipped_file_path was: {}", unzipped_file_path);
+
+            //lazy comparison assumes files all contain the same string literal
+            assert_eq!("here's some file contents", unzipped_contents);
+            //clear contents of temp file to be written to again with .copy_to()
+            //tmpfile.set_len(0);
+        }
```

Comment:
This all seems quite complicated. I don't know what else you tried, of course. I'd probably make a `HashMap<String, String>` that defines the files I want to write:

```rust
let files: HashMap<String, String> = (0..10)
    .map(|i| (format!("/file{}", i), "test".to_string()))
    .collect();
```

Then, you can write these to files, and build the archive, just as you do above. And here, where you want to test that the server responds with the correct content, you can iterator over the map of `(path, content)` values again, and directly compare the content with `reqwest::get(&format!("http://localhost:3000{}", path))?.text()?`.